homeassistant/components/sonos/speaker.py

```diff
@@ -365,6 +380,42 @@ async def async_update_device_properties(self, event: SonosEvent = None) -> None
 
         self.async_write_entity_states()
 
+    def update_alarms_for_speaker(self) -> set[str]:
+        """Update current alarm instances.
+
+        Updates hass.data[DATA_SONOS].alarms and returns a list of all alarms that are new.
+        """
+        new_alarms = set()
+        stored_alarms = self.hass.data[DATA_SONOS].alarms
+        updated_alarms = get_alarms(self.soco)
+
+        for alarm in updated_alarms:
+            if alarm.zone.uid == self.soco.uid and alarm.alarm_id not in list(
+                stored_alarms.keys()
+            ):
+                new_alarms.add(alarm.alarm_id)
+                stored_alarms[alarm.alarm_id] = alarm
+
+        for alarm_id, alarm in list(stored_alarms.items()):
```

Comment:
```suggestion
        for alarm_id, alarm in stored_alarms.items():
```

Reply:
Thats unfortunately not possible because you get an error if you edit a dict while looping over it :D

Reply:
Yep, I didn't notice that, good catch 👍